File: day2/first/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	input, err := os.Open("../input.txt")
	if err != nil {
		os.Exit(1)
	}
	scanner := bufio.NewScanner(input)

	safe := 0

Outer:
	for scanner.Scan() {
		report := scanner.Text()
		levelsStr := strings.Split(report, " ")

		levels := []int{}
		for _, level := range levelsStr {
			num, _ := strconv.Atoi(level)
			levels = append(levels, num)
		}

		isDecreasing := false
		if levels[1] < levels[0] {
			isDecreasing = true
		}

		for i := 1; i < len(levels); i++ {
			if isDecreasing {
				if levels[i] < levels[i-1]-3 || levels[i] >= levels[i-1] {
					continue Outer
				}
			} else {
				if levels[i] > levels[i-1]+3 || levels[i] <= levels[i-1] {
					continue Outer
				}
			}
		}
		safe += 1
	}
	fmt.Println(safe)
}

File: day2/first/main.py
safe = 0

with open("../input.txt") as file:
    while (line := file.readline()):
        levels = line.split()
        levels = [int(level) for level in levels]

        is_decreasing = False
        if levels[0] > levels[1]:
            is_decreasing = True
        
        for i in range(1, len(levels)):
            if is_decreasing:
                if levels[i] >= levels[i-1] or levels[i] < levels[i-1] - 3:
                    break
            else:
                if levels[i] <= levels[i-1] or levels[i] > levels[i-1] + 3:
                    break
        else:
            safe += 1
    
print(safe)